File: haushaltsbuch/src/main/java/org/github/haschi/haushaltsbuch/api/_Buchung.java
package org.github.haschi.haushaltsbuch.api;

import org.apache.commons.lang3.StringUtils;
import org.github.haschi.haushaltsbuch.infrastruktur.modellierung.de.Information;

@Information
public interface _Buchung
{
    String buchungstext();

    Währungsbetrag betrag();

    static Buchung leer()
    {
        return Buchung.builder()
                .buchungstext(StringUtils.EMPTY)
                .betrag(_Währungsbetrag.NullEuro())
                .build();
    }
}

File: haushaltsbuch/src/main/kotlin/org/github/haschi/haushaltsbuch/api/Buchung.kt
package org.github.haschi.haushaltsbuch.api

data class Buchung(val buchungstext: String, val betrag: Währungsbetrag) {

    companion object {

        val leer: Buchung
        get() = Buchung(String.empty, Währungsbetrag.NullEuro)
    }
}

val String.Companion.empty: String
    get() = ""
